diff -r -N -C2 gas-980103.orig/libiberty/config/mt-mingw32 gas-980103/libiberty/config/mt-mingw32
*** gas-980103.orig/libiberty/config/mt-mingw32	Thu Jan  1 00:00:00 1970
--- gas-980103/libiberty/config/mt-mingw32	Sat Jan  3 15:27:05 1998
***************
*** 0 ****
--- 1,27 ----
+ # Mingw32 target Makefile fragment.
+ # The autoconfiguration fails for a Mingw32 target, because of an
+ # incompatible definition of sys_errlist, which is imported from a DLL.
+ # Therefore, we compute the dependencies by hand.
+ 
+ HDEFINES = -DNO_SYS_PARAM_H -DNO_SYS_FILE_H
+ CONFIG_H = mgconfig.h
+ NEEDED_LIST = mgneeded-list
+ 
+ mgconfig.h: Makefile
+ 	if [ -f ../newlib/Makefile ]; then \
+ 	  $(MAKE) $(FLAGS_TO_PASS) xconfig.h; \
+ 	  cp xconfig.h mgconfig.h; \
+ 	else \
+ 	  echo "#define NEED_basename 1"    > mgconfig.h; \
+ 	  echo "#define NEED_sys_siglist 1" >> mgconfig.h; \
+ 	  echo "#define NEED_strsignal 1"   >> mgconfig.h; \
+ 	  echo "#define NEED_psignal 1"     >> mgconfig.h; \
+ 	fi
+ 
+ mgneeded-list: Makefile
+ 	if [ -f ../newlib/Makefile ]; then \
+ 	  $(MAKE) $(FLAGS_TO_PASS) xneeded-list; \
+ 	  cp xneeded-list mgneeded-list; \
+ 	else \
+ 	  echo getopt.o vasprintf.o > mgneeded-list; \
+ 	fi
diff -r -N -C2 gcc-2.95.2/libiberty/config.table~ gcc-2.95/libiberty/config.table
*** gcc-2.95.2/libiberty/config.table~	Wed Apr  7 05:31:23 1999
--- gcc-2.95.2/libiberty/config.table	Mon Jul 31 21:01:49 2000
***************
*** 23,26 ****
--- 23,30 ----
      arm*-*-*)		frags="${frags} ../../config/mh-armpic" ;;
      hppa*-*-*)		frags="${frags} ../../config/mh-papic" ;;
+     i[3456]86-*-mingw32*)	frags="${frags} ../../config/mt-mingw32" ;;
+      # Mingw32 does not require strerror.o from REQUIRED_OFILES.
+      # Worse: it will not compile it because of an incompatible sys_errlist
+      # definition.
      i[3456]86-*-*)	frags="${frags} ../../config/mh-x86pic" ;;
      powerpc*-*-aix*)	;;
diff -r -N -C2 gas-980103.orig/libiberty/pexecute.c gas-980103/libiberty/pexecute.c
*** gas-980103.orig/libiberty/pexecute.c	Sat Jan  3 09:12:16 1998
--- gas-980103/libiberty/pexecute.c	Sat Jan  3 15:44:23 1998
***************
*** 223,227 ****
--- 223,230 ----
  #if defined (_WIN32)
  
+ #include <io.h>
+ #include <fcntl.h>
  #include <process.h>
+ #include <signal.h>
  extern int _spawnv ();
  extern int _spawnvp ();
***************
*** 236,240 ****
     to remove the outermost set of double quotes from all arguments.  */
  
! const char * const *
  fix_argv (argvec)
       char **argvec;
--- 239,243 ----
     to remove the outermost set of double quotes from all arguments.  */
  
! char * const *
  fix_argv (argvec)
       char **argvec;
***************
*** 253,257 ****
            if (temp[j] == '"')
              {
!               newtemp = xmalloc (len + 2);
                strncpy (newtemp, temp, j);
                newtemp [j] = '\\';
--- 256,260 ----
            if (temp[j] == '"')
              {
!               newtemp = (char *) xmalloc (len + 2);
                strncpy (newtemp, temp, j);
                newtemp [j] = '\\';
***************
*** 267,271 ****
        }
  
!   return (const char * const *) argvec;
  }
  
--- 270,274 ----
        }
  
!   return (char * const *) argvec;
  }
  
