--- gcc-13.4.0/libstdc++-v3/config/io/basic_file_stdio.cc.orig	2025-06-05 18:03:03.000000000 +0200
+++ gcc-13.4.0/libstdc++-v3/config/io/basic_file_stdio.cc	2026-04-07 20:44:23.050840761 +0200
@@ -251,7 +251,7 @@ _GLIBCXX_BEGIN_NAMESPACE_VERSION
     const char* __c_mode = fopen_mode(__mode);
     if (__c_mode && !this->is_open())
       {
-#ifdef _GLIBCXX_USE_LFS
+#if defined(_GLIBCXX_USE_LFS) || defined(__MINGW32__)
 	if ((_M_cfile = fopen64(__name, __c_mode)))
 #else
 	if ((_M_cfile = fopen(__name, __c_mode)))
@@ -417,7 +417,7 @@ _GLIBCXX_BEGIN_NAMESPACE_VERSION
 	  return -1;
       }
     return __way == ios_base::beg ? __off : std::get_file_offset(this);
-#elif defined(_GLIBCXX_USE_LFS)
+#elif defined(_GLIBCXX_USE_LFS) || defined(__MINGW32__)
     return lseek64(this->fd(), __off, __way);
 #else
     if (__off > numeric_limits<off_t>::max()
@@ -463,6 +463,15 @@ _GLIBCXX_BEGIN_NAMESPACE_VERSION
 	const streamoff __off = __buffer.st_size - std::get_file_offset(this);
 	return std::min(__off, streamoff(numeric_limits<streamsize>::max()));
       }
+#elif defined(__MINGW32__)
+    struct _stati64 __buffer;
+    const int __err = _fstati64(this->fd(), &__buffer);
+    if (!__err && _GLIBCXX_ISREG(__buffer.st_mode))
+      {
+	const streamoff __off = __buffer.st_size - lseek64(this->fd(), 0,
+							   ios_base::cur);
+	return std::min(__off, streamoff(numeric_limits<streamsize>::max()));
+      }
 #else
     struct stat __buffer;
     const int __err = fstat(this->fd(), &__buffer);
diff -r faddb1a6e933 libstdc++-v3/include/ext/stdio_sync_filebuf.h
--- a/libstdc++-v3/include/ext/stdio_sync_filebuf.h	Fri Jul 31 08:58:42 2009 -0600
+++ b/libstdc++-v3/include/ext/stdio_sync_filebuf.h	Fri Jul 31 13:09:27 2009 -0600
@@ -163,7 +163,7 @@
 	  __whence = SEEK_CUR;
 	else
 	  __whence = SEEK_END;
-#ifdef _GLIBCXX_USE_LFS
+#if defined(_GLIBCXX_USE_LFS) || defined(__MINGW32__)
 	if (!fseeko64(_M_file, __off, __whence))
 	  __ret = std::streampos(ftello64(_M_file));
 #else
