--- a/libstdc++-v3/include/ext/stdio_sync_filebuf.h	Fri Jul 31 08:58:42 2009 -0600
+++ b/libstdc++-v3/include/ext/stdio_sync_filebuf.h	Fri Jul 31 13:09:27 2009 -0600
@@ -163,7 +163,7 @@
 	  __whence = SEEK_CUR;
 	else
 	  __whence = SEEK_END;
-#ifdef _GLIBCXX_USE_LFS
+#if defined(_GLIBCXX_USE_LFS) || defined(__MINGW32__)
 	if (!fseeko64(_M_file, __off, __whence))
 	  __ret = std::streampos(ftello64(_M_file));
 #else
